crystalball/wsclean.py

```diff
@@ -114,6 +175,44 @@ def import_from_wsclean(wsclean_comp_list, include_regions=[],
                          in wsclean_comps.items()}
         log.info("Selecting up %d brightest sources", num)
 
+    if clean_mask_file:
+        header, clean_mask = load_clean_mask(clean_mask_file)
+        wcs = WCS(header)
+
+        ra = np.rad2deg(wsclean_comps["Ra"]) * u.rad
+        dec = np.rad2deg(wsclean_comps["Dec"]) * u.rad
+        flux = wsclean_comps["I"]
+
+        freq = SpectralQuantity(wsclean_comps["ReferenceFrequency"] * u.Hz,
+                                doppler_rest=wcs.wcs.restfrq * u.Hz,
+                                doppler_convention="radio")
+        x, y, z = wcs.wcs_world2pix(ra, dec, freq.to(u.m/u.s), 0)
+        x, y, z = np.rint([x, y, z]).astype(int)
+
+        log.info("Grouping sources by id's in %s", clean_mask_file)
+
+        source_ids = clean_mask[z, y, x]
+
+        if(np.any(source_ids == 0)):
+            mask = np.where(source_ids == 0)[0]
+            raise ValueError(f"Unable to register sources {wsclean_comps['Name'][mask]}")
+
+        nr_sources = clean_mask.max()
+        source_id_range = np.arange(1, nr_sources + 1)
+
+        integrated_fluxes = np.array([flux[sid == source_ids].sum() for sid in source_id_range])
+        broadcast_fluxes = integrated_fluxes[source_ids - 1]
+
+        comp_sort_idx = np.lexsort((flux, broadcast_fluxes))[::-1]
```

Comment:
Broadcast the total flux for each source over the number of a components and lexically sort. This allows us to partially select components of a source.